src/java/org/apache/cassandra/rocksdb/RocksDBEngine.java

```diff
@@ -221,49 +266,6 @@ public AbstractStreamReceiveTask getStreamReceiveTask(StreamSession session, Str
         return new RocksDBStreamReceiveTask(session, summary.cfId, summary.files, summary.totalSize);
     }
 
-    private void applyRowToRocksDB(ColumnFamilyStore cfs,
-                                   boolean writeCommitLog,
-                                   DecoratedKey partitionKey,
-                                   UpdateTransaction indexer,
-                                   Row row)
-    {
-
-        Clustering clustering = row.clustering();
-
-        byte[] rocksDBKey = RowKeyEncoder.encode(partitionKey, clustering, cfs.metadata);
-        byte[] rocksDBValue = RowValueEncoder.encode(cfs.metadata, row);
-
-        try
-        {
-            indexer.start();
-            rocksDBFamily.get(cfs.metadata.cfId).merge(partitionKey, rocksDBKey, rocksDBValue);
-            if (indexer != UpdateTransaction.NO_OP)
-            {
-                try
-                {
-                    secondaryIndexMetrics.rsiTotalInsertions.inc();
-                    indexer.onInserted(row);
-                }
-                catch (RuntimeException e)
-                {
-                    secondaryIndexMetrics.rsiInsertionFailures.inc();
-                    logger.error(e.toString(), e);
-                    throw new StorageEngineException("Index update failed", e);
-                }
-
-            }
-        }
-        catch (RocksDBException e)
-        {
-            logger.error(e.toString(), e);
-            throw new StorageEngineException("Row merge failed", e);
-        }
-        finally
-        {
-            indexer.commit();
```

Comment:
I'm not sure about this. Why do we commit the indexer when we've experienced an exception?